front/lib/resources/data_source_resource.ts

```diff
@@ -376,7 +376,7 @@ export class DataSourceResource extends ResourceWithVault<DataSource> {
   }
 
   canWrite(auth: Authenticator) {
-    return auth.isBuilder() && auth.canWrite([this.vault.acl()]);
+    return auth.canWrite([this.vault.acl()]);
```

Comment:
now that we have some logic in vault.canRead, we need to delegate to vault here . so : 
```suggestion
    return this.vault.canWrite(auth);
```
same for canRead, and the same methods in data_source_view_resource